src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -105,9 +113,22 @@ public PrivilegesEvaluatorResponse evaluate(
         final Task task,
         final String action,
         final Resolved requestedResolved,
-        final PrivilegesEvaluatorResponse presponse
+        final PrivilegesEvaluatorResponse presponse,
+        ConfigModelV7.SecurityRoles securityRoles
     ) {
+
         final boolean isDebugEnabled = log.isDebugEnabled();
+
+        if (matchAnySystemIndices(requestedResolved) && !checkSystemIndexPermissionsForUser(securityRoles)) {
+            log.warn(
+                "An account without the {} permission  is trying to access a System Index. Related indexes: {}",
+                ConfigConstants.SYSTEM_INDEX_PERMISSION,
+                getProtectedIndexes(requestedResolved).stream().collect(Collectors.joining(", "))
```

Comment:
Guard the info message in `log.isInfoEnabled()` so we don't take the perf hit assembling the message even if it isn't used